crates/rover-client/src/operations/persisted_queries/publish/types.rs

```diff
@@ -152,3 +287,170 @@ impl PersistedQueriesOperationCounts {
+
+        let relay_manifest: RelayPersistedQueryManifest =
+            serde_json::from_str(&relay_manifest).expect("could not read relay manifest");
+        let apollo_manifest: ApolloPersistedQueryManifest = relay_manifest.try_into().unwrap();
+        assert_eq!(
+            apollo_manifest.operations[0],
+            PersistedQueryOperation {
+                name: "NewsfeedQuery".to_string(),
+                r#type: PersistedQueryOperationType::Query,
+                id: id.to_string(),
+                body: body.to_string()
+            }
+        );
+    }
+
+    #[test]
+    fn it_can_convert_relay_manifest_with_multiple_queries_to_apollo_manifest() {
+        let id_one = "ed145403db84d192c3f2f44eaa9bc6f9";
+        let body_one = "query NewsfeedQuery {\n  topStory {\n    title\n    summary\n    poster {\n      __typename\n      name\n      profilePicture {\n        url\n      }\n      id\n    }\n    thumbnail {\n      url\n    }\n    id\n  }\n}\n";
+        let id_two = "adkjflaskdjf";
+        let body_two = "mutation NamedMutation { topStory }";
+        let relay_manifest = serde_json::json!({id_one: body_one, id_two: body_two}).to_string();
+
+        let relay_manifest: RelayPersistedQueryManifest =
+            serde_json::from_str(&relay_manifest).expect("could not read relay manifest");
+        let mut apollo_manifest: ApolloPersistedQueryManifest = relay_manifest.try_into().unwrap();
+
+        // guarantee proper ordering
+        apollo_manifest.operations.sort();
+
+        assert_eq!(
+            apollo_manifest.operations[0],
+            PersistedQueryOperation {
+                name: "NamedMutation".to_string(),
+                r#type: PersistedQueryOperationType::Mutation,
+                id: id_two.to_string(),
+                body: body_two.to_string()
+            }
+        );
+
+        assert_eq!(
+            apollo_manifest.operations[1],
+            PersistedQueryOperation {
+                name: "NewsfeedQuery".to_string(),
+                r#type: PersistedQueryOperationType::Query,
+                id: id_one.to_string(),
+                body: body_one.to_string()
+            }
+        );
+    }
+
+    #[test]
+    fn relay_manifest_with_anonymous_operations_fails() {
+        let relay_manifest = r#"{
+      "ed145403db84d192c3f2f44eaa9bc6f9": "query {\n  topStory {\n    title\n    summary\n    poster {\n      __typename\n      name\n      profilePicture {\n        url\n      }\n      id\n    }\n    thumbnail {\n      url\n    }\n    id\n  }\n}\n"
+    }"#;
+
+        let relay_manifest: RelayPersistedQueryManifest =
+            serde_json::from_str(relay_manifest).expect("could not read relay manifest");
+        let apollo_manifest_result: Result<ApolloPersistedQueryManifest, RoverClientError> =
+            relay_manifest.try_into();
+        assert!(matches!(
+            apollo_manifest_result,
+            Err(RoverClientError::RelayOperationParseFailures { .. })
```

Comment:
I think it would be reasonable to test the actual error message (or that it has an appropriate substring or something) in this and other cases.